src/sql/arrow_sql_gen/mysql.rs

```diff
@@ -275,6 +273,10 @@ pub fn rows_to_arrow(rows: &[Row]) -> Result<RecordBatch> {
                         i
                     );
                 }
+                column_type @ (ColumnType::MYSQL_TYPE_STRING
```

Comment:
1. Is this for acceleration? 
2. Could you please verify that query returns the same result as direct query
3. As we are changing to MYSQL_TYPE_STRING to be binary I propose to change  MYSQL_TYPE_STRING->MYSQL_TYPE_VARCHAR for `char`
_ if data_type.starts_with("char") => ColumnType::MYSQL_TYPE_STRING,

Reply:
Example query
![image](https://github.com/user-attachments/assets/e30fc3ea-23da-4162-9d70-7c56c086dfed)

Reply:
This is for acceleration, the verified data can be seen from spice oss ticket

Reply:
Checked with the new fix, it doesn't broke query in spice repl, see the query result in: https://github.com/spiceai/spiceai/issues/1871

Reply:
Added new code to detect whether the column is binary (BINARY, VARBINARY) or non binary (CHAR VARCHAR) . The change don't have any degrading behavior tested on tpch queries